internal/cloudns/resource_dns_record.go

```diff
@@ -138,6 +146,58 @@ func resourceDnsRecordDelete(ctx context.Context, d *schema.ResourceData, meta i
 	return resourceDnsRecordRead(ctx, d, meta)
 }
 
+func resourceImportStateContext(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	config := meta.(ClientConfig)
+	tflog.Debug(ctx, fmt.Sprintf("IMPORT ID: %#v", d.Id()))
+
+	// Extract zone name and record id
+	halves := strings.Split(d.Id(), "/")
+	interestingHalf := halves[1]
+	parts := strings.Split(interestingHalf, "_")
+	if len(parts) != 3 {
+		errMsg := fmt.Sprintf("Could not extract zone from record ID: %s", d.Id())
+		tflog.Error(ctx, errMsg)
+		return nil, errors.New(errMsg)
+	}
+	zone := fmt.Sprintf("%s.%s", parts[0], parts[1])
+	wantedID := parts[2]
+
+	tflog.Debug(ctx, fmt.Sprintf("Trying to import record %s from zone: %s", wantedID, zone))
+
+	config.rateLimiter.Take()
+	zoneRead, err := cloudns.Zone{Domain: zone}.List(&config.apiAccess)
+	if err != nil {
+		return nil, err
+	}
+
+	recordFound := false
+
+	// Try to find record by ID from the zone record list.
+	// @todo There is a more direct way to doing this with search parameters.
+	for _, zoneRecord := range zoneRead {
+		actualId := zoneRecord.ID
+		//tflog.Debug(ctx, fmt.Sprintf("wantedID: %s, actualID: %s", wantedID, actualId))
+		if wantedID == actualId {
+			err = updateState(d, &zoneRecord)
+			if err != nil {
+				tflog.Error(ctx, fmt.Sprintf("IMPORT error updateState(): %s", err))
+				return nil, err
+			}
+			tflog.Debug(ctx, fmt.Sprintf("IMPORT %s.%s %d in %s %s", zoneRecord.Host, zoneRecord.Domain, zoneRecord.TTL, zoneRecord.Rtype, zoneRecord.Record))
+			recordFound = true
```

Comment:
Thanks for the PR.  I have been using this in my fork for some time. At first I ran into following issue.

Records are imported with ID `$subdmain_$zone_$recordId`, but actual record ID in tfstate should be `$recordId` after import. This ID will be used in future API requests, after we start working with imported records.

This will fix the issue:
```suggestion
			recordFound = true
			d.SetId(actualId)
```